### PR TITLE
sidebar

### PR DESCRIPTION
**Pull request by Prabhatm09**
work -: sidebar and navbar

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,20 @@
 import './App.css';
```
Comment (Souvikdeb2612) [inline src/App.js]: First external imports should occur then 1 line space, then internal files and component imports and then 1 line space and finally css imports

File: src/components/home/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import "./Home.css"
```
Comment (Souvikdeb2612) [inline src/components/home/Home.jsx]: format imports like above

File: src/components/navbar/Navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import "./Navbar.css"
```
Comment (Souvikdeb2612) [inline src/components/navbar/Navbar.jsx]: space after import

File: src/components/navbar/Navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
...uqpeUlpUM+GGySe5nE8wrXy5IgMkHIMeWuqVbY9MgRtynygqikbzHJaenB4ONZ4wrC28KlT+0PQscATOb+rVvLk1a7l3Y7mWO9tXeocZIO+BI8WDByfDIbtmTWR+ijTkNQhFLkHVxy48t/7R/yFVpUaYLMW7dzF09NvalcUrWyrVqoXP3SAB55lz4N4er2rm91SjTSsBilT+9yes7hXKbJ2WxrQ64N0Krp/Pd3I5XdcIvcCWZjOM8E7zfGPFYeVZNzlrOs0EzTzNBM8ZwdYwbNEs8QWgAotE5iC0TzRoYoNFBo3DRQeMMHKtCK8aB4RX9YBg9V5H6jxLpmnErVuBUqL1p0vePz8pS+JeJa9atVtbKpyUFPKWTYv57+Uq25OSSZzpWNW/Jeb3+IdUhlsbJF8mrHP8h/eVXU+INW1I4ur2sU6hKZ5F+g6/OMMZnuX0ibKKuKDaTci31CnVrboThie2e8vwp8wGNx2I3zM6xLrwdqK3KDT65+2UfZE/iUdviJKzcLQXZJrSzONbk9jiTK2ozgiFNqmOkyORoUSDo6aKjgsu3wkrT0miyrmmMjoY9pUFXtHlNNthOG3pTOgNrbGknKNhBXS6igY2ns1Q9kq8y5+Yz/SSSJF+GJ3G2UfhkZUQl8opd3xVV0yr4Wt6XcW2fu1UIem3/L+8eWnEGnXhCpcKjn8FX3TH3FFGlUtX8VVKCmebImSk4GPwgADM2U2ua7MtvjxXway1TyMEzym8Pa61HFrcuWpHZWJyU/aWo1M9OkuZZRcQpeDLwZeILesBBC8TzwTNEc8Y8CB94oNG2YtSYDwcB4w4hvzZaRXqKcOw5F+f7ZjkGVrjiqwo2tLPuMzEj1AibwcVrKqDljk9YoHeDp7g+hipM0isjpPdpzM6u8APYhqFR6VRalNylRCGVh2MRjGJ7vFgzUuF9epaxb8lUqt3THvr+b1EneQGYta3Na2qpXt3NOojDlZe017h28q6hpVvc3HL4jjflGB1mS6vj2jTXPemP0pbxwi4nl6xcgW070ncgxLTvRDART/AOIuoLa2PgIc1K45fgO8zGqxCAZxneT3HF3WuOIa61WyKRCIPIYzIGv94TdUsgZbHsgivysB5bS76Jee0afTyffQYP6Si/6v/KWPhV2+1TO2M/z/AHl4sz2LoshaILxLGIJnZBIUXieeDYxOTAeH/9k="
```
Comment (Souvikdeb2612) [inline src/components/navbar/Navbar.jsx]: is this image constant or will it change based on API?

File: src/components/sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                <span>Dasboard</span>
+            </li>
+            <p className="title">User</p>
+            <li>
+                <span>Logout</span>
+            </li>
+        </ul>
+    </div>
+    <div className="bottom">
+        <div className="colorchanger"></div>
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: make this self closing

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/login/Registration.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const data = new FormData(e.currentTarget);
+        const actualData = {
+            email: data.get("email"),
+            password: data.get("password"),
+        }
+        
+        if(actualData.email && actualData.password){
+            console.log(actualData)
```
Comment (Souvikdeb2612) [inline src/components/login/Registration.jsx]: remove/comment console statements before pushing

File: src/components/login/Registration.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+              document.getElementById('login-form').reset()
```
Comment (Souvikdeb2612) [inline src/components/login/Registration.jsx]: Why are you directly manipulating the DOM?

File: src/components/login/UserLogin.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const data = new FormData(e.currentTarget);
+        const actualData = {
+            email: data.get("email"),
+            password: data.get("password"),
+        }
+        
+        if(actualData.email && actualData.password){
+            console.log(actualData)
```
Comment (Souvikdeb2612) [inline src/components/login/UserLogin.jsx]: same as above

File: src/components/navbar/Navbar.jsx [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import InputBase from '@mui/material/InputBase';
+import Badge from '@mui/material/Badge';
+import MenuItem from '@mui/material/MenuItem';
+import Menu from '@mui/material/Menu';
+import MenuIcon from '@mui/icons-material/Menu';
+import SearchIcon from '@mui/icons-material/Search';
+import AccountCircle from '@mui/icons-material/AccountCircle';
+import MailIcon from '@mui/icons-material/Mail';
```
Comment (Souvikdeb2612) [inline src/components/navbar/Navbar.jsx]: Fix import format

File: src/components/navbar/Navbar.jsx [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+    paddingLeft: `calc(1em + ${theme.spacing(4)})`,
+    transition: theme.transitions.create('width'),
+    width: '100%',
+    [theme.breakpoints.up('md')]: {
+      width: '20ch',
+    },
+  },
+}));
+
+export const Navbar = (props) => {
```
Comment (Souvikdeb2612) [inline src/components/navbar/Navbar.jsx]: destructure props here

File: src/components/navbar/Navbar.jsx [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+            <StyledInputBase
+              placeholder="Search…"
+              inputProps={{ 'aria-label': 'search' }}
+            />
+          </Search>
+          <Box sx={{ flexGrow: 1 }} />
+          <Box sx={{ display: { xs: 'none', md: 'flex' } }}>
+            <IconButton size="large" aria-label="show 4 new mails" color="inherit">
+              <Badge  color="error">
+                <DarkModeOutlinedIcon className="icons" onClick={()=> dispatch({type: "TOGGLE"})} />
```
Comment (Souvikdeb2612) [inline src/components/navbar/Navbar.jsx]: create a function and call here

File: src/components/sidebar/Sidebar.jsx [hunk 12]
```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: Fix import format

File: src/components/sidebar/Sidebar.jsx [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import InboxIcon from '@mui/icons-material/MoveToInbox';
+import MailIcon from '@mui/icons-material/Mail';
+import { Navbar } from "../navbar/Navbar"
+import { NavLink } from "react-router-dom"
+
+const drawerWidth = 240;
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx:26]: export constants from separate constant files

File: src/components/sidebar/Sidebar.jsx [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+      '& .MuiDrawer-paper': closedMixin(theme),
+    }),
+  }),
+);
+
+export const Sidebar = () => {
+  const theme = useTheme();
+  const [open, setOpen] = React.useState(false);
+
+  const handleDrawerOpen = () => {
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: Write it like this:   const handleDrawerOpen =()=>setOpen(true)

File: src/components/sidebar/Sidebar.jsx [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+    setOpen(true);
+  };
+
+  const handleDrawerClose = () => {
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: same as above

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/layout/Layout.jsx [hunk 16]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { styled, ThemeProvider, createTheme } from "@mui/material/styles";
+import CssBaseline from "@mui/material/CssBaseline";
+import MuiAppBar from "@mui/material/AppBar";
+
+import { Navbar } from "../navbar/Navbar";
+import { Sidebar } from "../sidebar/Sidebar";
+
+const drawerWidth = 240;
```
Comment (Souvikdeb2612) [inline src/components/layout/Layout.jsx:10]: export constants from a separate constant file

File: src/components/login/Registration.jsx [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const data = new FormData(e.currentTarget);
+    const actualData = {
+      email: data.get("email"),
+      password: data.get("password"),
+    };
+
+    if (actualData.email && actualData.password) {
+      console.log(actualData);
```
Comment (Souvikdeb2612) [inline src/components/login/Registration.jsx]: remove/comment console statements before pushing

File: src/components/login/UserLogin.jsx [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const data = new FormData(e.currentTarget);
+    const actualData = {
+      email: data.get("email"),
+      password: data.get("password"),
+    };
+
+    if (actualData.email && actualData.password) {
+      console.log(actualData);
```
Comment (Souvikdeb2612) [inline src/components/login/UserLogin.jsx]: same as above

File: src/components/sidebar/Sidebar.jsx [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+    ...openedMixin(theme),
+    "& .MuiDrawer-paper": openedMixin(theme),
+  }),
+  ...(!open && {
+    ...closedMixin(theme),
+    "& .MuiDrawer-paper": closedMixin(theme),
+  }),
+}));
+
+export const Sidebar = (props) => {
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: destructure props here

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/login/Login.jsx [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
...eR3Ly3vkVKjLSucANTY7Ic49qkT7QO/o+0OscCdfpywB4wNsomrVvpe5pgCnc10A4Ba1RQPINicXGlbmoMVLis47HrVGHoWxBpshf8AKK1oMtOrXRajMqKm0Gcs7BVGwuW4kb8YHXMvNYuSFEG+s1xu5+kcY7HB/dNnYQREQEREBERAREQPmVVpjksU0nTZqL1LWvU2jshiFZs7Qcj2QGO11DZz2GWtAhXfHF9beSwsKdugpUUCoMkKOGScn4yndY1Epf1ifnqjjw2FT80Mu2VtrZ0OzCneIM7AKVMDgudpWP0QSwP1hCrk06seo9IfyLDNe24QFiHBOOpRnaJ7ABLW5e0S9hcAdSh/JGVz8AZ7eTS0vk1BqCBEamrAYxxUHf2njv6+MyVamrqyMMqwIYHgQRgg+UGLD00mu/KheT+ga187JR2QEALMxIVQTgcASScHAx1GWVoDV/QoEPXPP1BvG0MIp7l35+0T4CdPJDkzcWF1XGVa2ddz56ZKt0AR2gM+eo/CTqNvODj1rG5juASLaZV+dY7Sq5b9mS5Qc0KDdENkZPPb2A6im0CMSVT5Izxh1sdobGyxT+iLZp8ANnZXJXHzS22Rjdg7t2Jk4iBCdZPJu50jToUrY0wEcu+2zLk7JVcbKtn2m+Er7/hPpH3rb/Fqf7cveIEN1ccmKujrerTrlDUeoX6DFlChFVRlgN+Qx4dcmURATz3lolZGpVUV0YYZWAII7wZ6IgVVpzVCjEtZVig/+uoCyjuVx0gPrBj3yJXWrPSacKKVPqVU9emVmwUQNc/+HulP+Ub/ABaH+5PfaartIvjaSlTB47dQEjyQNky/IhO1WaD1SCm6Vbi5LFGDBKaADKkMAWbORke6JakRCCeDSui6N3TajcIroeo9R6ipG9WHaMGe+IFRaa1QHJazrjZ6krA7u4VEHDxXPeZF7nVtpNDgUFfvWrTx+NlPwmwsQNdBq90p/wAo3+LQ/wByZK01VaQc9PmaY+lULH0RT+cviIFc8ltV62lanc1rhqlSmdpVRAiZwR0tosW49WzLHiICIiAiIgIiICIiAiIgJxOYgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgIiICIiAiIgf/Z"
```
Comment (devmanit907) [inline src/components/login/Login.jsx]: Use this image in jpg/png format

File: src/components/login/Login.jsx [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+                <UserLogin />
+              </TabPanel>
+              <TabPanel value={value} index={1}>
+                <Registration />
+              </TabPanel>
+            </Box>
+          </Card>
+        </Grid>
+        <Grid sx={{ backgroundColor: "#291646" }}>
+          {/* <Avatar   sx={{ width: 600, height: 556 , marginLeft:30}}  variant="square" alt="Remy Sharp" src="https://resources.pulse.icc-cricket.com/photo-resources/2022/10/13/037e8b5a-9f7b-4ee9-aa3d-b390aa0d1445/Main-Image-SSO.png?height=1024&width=591" /> */}
```
Comment (devmanit907) [inline src/components/login/Login.jsx]: Remove commented code, if not used

File: src/components/login/Registration.jsx [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+      setError({
+        status: true,
+        msg: "Registraction successfull",
+        type: "success",
+      });
+      //   navigate('/')
+    } else {
+      setError({
+        status: true,
+        msg: "confrim passowrd dosen't match",
```
Comment (devmanit907) [inline src/components/login/Registration.jsx]: Confirm spelling is wrong
Doesn't spelling is wrong

File: src/components/login/Registration.jsx (hunk 6 above)
Comment (Prabhatm09) [reply src/components/login/Registration.jsx]: removed the console

File: src/components/login/UserLogin.jsx (hunk 8 above)
Comment (Prabhatm09) [reply src/components/login/UserLogin.jsx]: removed

File: src/components/navbar/Navbar.jsx (hunk 9 above)
Comment (Prabhatm09) [reply src/components/navbar/Navbar.jsx]: solved the formats

File: src/components/login/Registration.jsx (hunk 7 above)
Comment (Prabhatm09) [reply src/components/login/Registration.jsx]: to blank the input

File: src/components/navbar/Navbar.jsx (hunk 11 above)
Comment (Prabhatm09) [reply src/components/navbar/Navbar.jsx]: its removed now

File: src/components/sidebar/Sidebar.jsx (hunk 14 above)
Comment (Prabhatm09) [reply src/components/sidebar/Sidebar.jsx]: solved

File: src/components/login/ResetPassword.jsx [hunk 23]
```diff
@@ -26,13 +26,13 @@ export function ResetPassword() {
...
             oldpassword: data.get("Oldpassword"),
             password: data.get("newpassword"),
-            password_confrimation: data.get("password_confrimation")
+            password_confirmation: data.get("password_confirmation")
 
         };
 
-        if (actualData.oldpassword && actualData.password && actualData.password_confrimation) {
+        if (actualData.oldpassword && actualData.password && actualData.password_confirmation) {
             console.log(actualData);
```
Comment (Souvikdeb2612) [inline src/components/login/ResetPassword.jsx]: remove/comment console statements

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

**Review by Souvikdeb2612 — APPROVED** [review]

**Pull request merged**